### PR TITLE
Replace sync client with meilisearch-python-async for our search endpoint

### PR DESCRIPTION
**Pull request by AndreasPB**
* Replaces the sync client with an async one for the search endpoint

### CODE REVIEW

File: backend/app/routers/search.py [hunk 1]
```diff
@@ -13,58 +13,48 @@
...
             user_input,
-            {
-                "limit": limit,
-                # This is using AND logic
-                "filter": [f'genres="{genre}"' for genre in genres],
-                "sort": ["popularity:desc"],
-            },
+            limit=limit,
+            sort=["popularity:desc"],
+            filter=[f'genres="{genre}"' for genre in genres],
```
Comment (nullxDEADBEEF) [inline backend/app/routers/search.py:49]: Plz give me back my comment :D

File: backend/app/routers/search.py [hunk 2]
```diff
@@ -13,58 +13,48 @@
...
-                "limit": limit,
-                # This is using OR logic
-                "filter": [
-                    [f'provider_names="{providers}"' for providers in providers]
-                ],
-                "sort": ["popularity:desc"],
-            },
+            limit=limit,
+            sort=["popularity:desc"],
+            filter=[[f'provider_names="{providers}"' for providers in providers]],
```
Comment (nullxDEADBEEF) [inline backend/app/routers/search.py:56]: Same as above :D

**Review by nullxDEADBEEF — CHANGES_REQUESTED** [review]

File: backend/app/routers/search.py (hunk 2 above)
Comment (AndreasPB) [reply backend/app/routers/search.py:59]: Done, and gave you an extra ;D

File: backend/app/routers/search.py (hunk 1 above)
Comment (AndreasPB) [reply backend/app/routers/search.py:51]: Done

File: backend/app/routers/search.py (hunk 2 above)
Comment (nullxDEADBEEF) [reply backend/app/routers/search.py:59]: Nice :raised_hands:

**Review by nullxDEADBEEF — APPROVED** [review]
:rocket: :shipit:

**Pull request merged**